tests/integration/solidity_contracts/UniswapV2/test_uniswap_v2_pair.py

```diff
@@ -0,0 +1,580 @@
+            mine_block(timestamp = block_timestamp + 10)
+            # swap to a new price eagerly instead of syncing
+            await pair.swap(
+                amount0Out=0,
+                amount1Out=expand_to_18_decimals(1),
+                to=owner.address,
+                data=b"",
+                caller_address=owner.starknet_address,
+            )
+
+            assert await pair.price0CumulativeLast() == initial_price[0]*10
+            assert await pair.price1CumulativeLast() == initial_price[1]*10
+            assert await pair.getResrves()[2] == block_timestamp + 10
+
+            await mine_block(block_timestamp + 20)
+            await pair.sync(caller_address=owner.starknet_address)
+
+            new_price = encode_price(expand_to_18_decimals(6), expand_to_18_decimals(2))
+            assert await pair.price0CumulativeLast() == initial_price[0]*10 + new_price[0]*10
+            assert await pair.price1CumulativeLast() == initial_price[1]*10 + new_price[1]*10
+            assert await pair.getResrves()[2] == block_timestamp + 20
+
+    @pytest.mark.skip("mine_block is not yet implemented")       
+    class TestFeeToOff:
+        @pytest.mark.parametrize(
+            "swap_amount, token_0_amount, token_1_amount, expected_output_amount",
+            [
+                [1, 5, 10, 1662497915624478906],
+                [1, 10, 5, 453305446940074565],
+                [2, 5, 10, 2851015155847869602],
+                [2, 10, 5, 831248957812239453],
+                [1, 10, 10, 906610893880149131],
+                [1, 100, 100, 987158034397061298],
+                [1, 1000, 1000, 996006981039903216],
+                [1, 5, 10, 997000000000000000],
+                [1, 10, 5, 997000000000000000],
+                [1, 5, 5, 997000000000000000],
+                [1003009027081243732, 5, 5, 1],
+            ],
+        )
+        async def test_should_check_if_feeToOff_works(swap_amount,self, pair, owner, mine_block,token_0_amount,
+            token_1_amount,
+            expected_output_amount):
+            pair, token_0, token_1 = pair
+            token_0_amount = expand_to_18_decimals(3)
+            token_1_amount = expand_to_18_decimals(3)
+            await token_0.transfer(
+                to=pair.evm_contract_address,
+                value=expand_to_18_decimals(token_0_amount),
+                caller_address=owner.starknet_address,
+            )
+            await token_1.transfer(
+                to=pair.evm_contract_address,
+                value=expand_to_18_decimals(token_1_amount),
+                caller_address=owner.starknet_address,
+            )
+            await pair.mint(owner.address, caller_address=owner.starknet_address)
+
+            await token_0.transfer(
+                to=pair.evm_contract_address,
+                value=expand_to_18_decimals(swap_amount),
+                caller_address=owner.starknet_address,
+            )
+            with kakarot_error("UniswapV2: K"):
```

Comment:
quite surprised by this error message, and also by the fact that the original test doesn't include an error assert